qt_ui/windows/QLiberationWindow.py

```diff
@@ -367,6 +371,32 @@ def onGameGenerated(self, game: Game):
         self.game = game
         GameUpdateSignal.get_instance().game_loaded.emit(self.game)
 
+    def onEndGame(self, state: TurnState):
+        result = None
+        if state == TurnState.WIN:
+            result = QMessageBox.information(
+                QApplication.focusWidget(),
+                "Victory!",
+                "You have won the campaign, do you wish to start a new one?",
+                QMessageBox.Yes,
+                QMessageBox.No,
+            )
+        elif state == TurnState.LOSS:
+            result = QMessageBox.information(
+                QApplication.focusWidget(),
+                "Defeat!",
+                "You have lost the campaign, do you wish to start a new one?",
+                QMessageBox.Yes,
+                QMessageBox.No,
+            )
+        for window in QApplication.topLevelWidgets():
+            if window is not self:
+                window.close()
+        sleep(0.5)
+        GameUpdateSignal.get_instance().updateGame(None)
+        if result is not None and result == QMessageBox.Yes:
+            self.newGame()
+
```

Comment:
End-game message, closing all windows except main, update game with `None` to prevent the user for manipulating the campaign further, start new campaign wizard if user chooses to start a new campaign.